in_progress/hd_shank_preprocessing.py

```diff
@@ -0,0 +1,148 @@
+    outside_channel_threshold: float = Field(default=-0.3, description="Threshold for outside channels")
+    n_neighbors: int = Field(default=11, description="Number of neighbors to use for bad channel detection")
+    seed: int = Field(default=0, description="Seed for random number generator")
+
+class CommonReferenceParameters(BaseModel):
+    reference: str = Field(default="global", description="Reference to use for common reference: 'global' or ...")
+    operator: str = Field(default="median", description="Operator to use for common reference: 'median' or ...")
+
+class HighpassSpatialFilterParameters(BaseModel):
+    n_channel_pad: int = Field(default=60, description="Number of channels to pad")
+    n_channel_taper: int = Field(default=None, description="Number of channels to taper")
+    direction: str = Field(default="y", description="Direction to use for highpass spatial filter: 'y' or ...")
+    apply_agc: bool = Field(default=True, description="Whether to apply automatic gain control")
+    agc_window_length_s: float = Field(default=0.01, description="Window length in seconds for automatic gain control")
+    highpass_butter_order: int = Field(default=3, description="Butterworth order for highpass filter")
+    highpass_butter_wn: float = Field(default=0.01, description="Butterworth wn for highpass filter")
+
+class HDShankPreprocessingParameters(BaseModel):
+    preprocessing_strategy: str = Field(default="cmr", description="Preprocessing strategy to use: destripe or cmr")
+    highpass_filter: HighpassFilterParameters = Field(default_factory=HighpassFilterParameters, description="Highpass filter parameters")
+    phase_shift: PhaseShiftParameters = Field(default_factory=PhaseShiftParameters, description="Phase shift parameters")
+    detect_bad_channels: DetectBadChannelsParameters = Field(default_factory=DetectBadChannelsParameters, description="Detect bad channels parameters")
+    remove_out_channels: bool = Field(default=True, description="Whether to remove out channels")
+    remove_bad_channels: bool = Field(default=True, description="Whether to remove bad channels")
+    max_bad_channel_fraction_to_remove: float = Field(default=0.5, description="Maximum fraction of bad channels to remove")
+    common_reference: CommonReferenceParameters = Field(default_factory=CommonReferenceParameters, description="Common reference parameters")
+    highpass_spatial_filter: HighpassSpatialFilterParameters = Field(default_factory=HighpassSpatialFilterParameters, description="Highpass spatial filter parameters")
+
+########################################################################################################################
+
+def hd_shank_preprocessing(
+    recording: si.BaseRecording,
+    params: HDShankPreprocessingParameters,
+    preprocessed_output_folder: Path,
+    verbose: bool = False
+):
+    recording_ps_full = spre.phase_shift(
+        recording,
+        margin_ms=params.phase_shift.margin_ms
+    )
```

Comment:
```suggestion
    if "inter_sample_shift" in recording.get_property_keys():
        recording_ps_full = spre.phase_shift(
            recording,
            margin_ms=params.phase_shift.margin_ms
        )
    else:
        recording_ps_full = recording
```
@magland we can make this optional, so that HD shanks that do not need a phase shift will not run it.